src/red_commons/logging/__init__.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from types import TracebackType
+from typing import Mapping, Optional, Tuple, Type, Union
+
+__all__ = ("RedTraceLogger", "VERBOSE", "TRACE", "maybe_update_logger_class", "getLogger")
+
+VERBOSE = logging.DEBUG - 3
+TRACE = logging.DEBUG - 5
+
+EXEC_INFO_TYPE = Union[
+    bool,
+    Tuple[Type[BaseException], BaseException, Optional[TracebackType]],
+    Tuple[None, None, None],
+    BaseException,
+]
+
+
+class RedTraceLogger(logging.Logger):
+    def __init__(self, name: str, level: int = logging.NOTSET) -> None:
+        super().__init__(name, level)
+
+        logging.addLevelName(VERBOSE, "VERBOSE")
+        logging.addLevelName(TRACE, "TRACE")
```

Comment:
I'm sorry I haven't noticed this earlier but these should be moved to `maybe_update_logger_class()` as they only really do anything on the first call.